src/main/java/duke/command/DeletePatientTaskCommand.java

```diff
@@ -66,31 +67,33 @@ public void execute(PatientTaskList patientTaskList, TaskManager tasks, PatientM
         if (patientId != 0) {
             try {
                 Patient toBeDeletedPatient = patientManager.getPatient(patientId);
-                for (PatientTask patientTask: patientTaskList.getPatientTask(patientId)) {
-                    if (patientTask.getUid() == taskId) {
-                        patientTaskList.deletePatientTaskByUniqueId(taskId);
-                        patientTaskStorage.save(patientTaskList.fullPatientTaskList());
-                        ui.patientTaskDeleted(patientTask, toBeDeletedPatient);
-                    }
+                if (patientTaskList.isPatientIdExist(patientId)) {
+                    patientTaskList.deleteEntirePatientTask(patientId);
+                    patientTaskStorage.save(patientTaskList.fullPatientTaskList());
+                    ui.patientTaskAllDeleted(patientManager.getPatient(patientId));
                 }
             } catch (DukeException e) {
-                throw new DukeException("Task or patient is not found!" + e.getMessage());
+                throw new DukeException("Patient is not found!");
```

Comment:
If patientId does not exist, the if statement will just skip. The DukeException will not be caught. Maybe instead of try and catch the exception, you can put else statement instead.

Reply:
noted